### PR TITLE
Iz gawb 1596 changes from kris doc

### PR DESCRIPTION
**Pull request by zarsky-broad**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -24,6 +24,7 @@
                        [uglify-js "git://github.com/mishoo/UglifyJS2#harmony"] ; override webpack's uglify with es6 compatible version
                        [url-loader "~0.5.7"]                ; webpack plugin
                        [webpack "~2.2.0"]                   ; combines deps into single js/css file
+                       [select2 "^4.0"]                     ; adds search to selects
```
Comment (MattPutnam) [inline project.clj]: Alphabetize

File: src/cljs/main/broadfcui/page/workspace/data/import_data.cljs [hunk 2]
```diff
@@ -47,21 +48,21 @@
       (if-let [result (:upload-result @state)]
         (if (:success? result)
           (style/create-flexbox {:style {:justifyContent "center" :paddingTop "1em"}}
-            (icons/icon {:style {:fontSize "200%" :color (:success-state style/colors)}} :done)
-            [:span {:style {:marginLeft "1em"}} "Success!"])
+                                (icons/icon {:style {:fontSize "200%" :color (:success-state style/colors)}} :done)
+                                [:span {:style {:marginLeft "1em"}} "Success!"])
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/data/import_data.cljs]: I had specifically aligned this "wrong" on purpose.  However, I don't want to fight with my editor.  Let's have IntelliJ indent this differently, and put that formatting into the master format.xml (or whatever that thing is).

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs [hunk 3]
```diff
@@ -88,14 +88,22 @@
...
-            :onChange (fn [event]
-                        (swap! state assoc :selected-workspace
-                          (nth workspaces-list (js/parseInt (.-value (.-target event))))))}
-           (map
-             (fn [ws] (str (get-in ws ["workspace" "namespace"]) "/" (get-in ws ["workspace" "name"])))
-             workspaces-list))])
+          {:value ""
+           :ref (utils/create-element-ref-handler
+                 {:store locals
+                  :key :workspace-select
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs:94]: Use a namespaced key (`::workspace-select`) to avoid collisions.

**Review by MattPutnam — APPROVED** [review]
👍 just a few minor things

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs (hunk 3 above)
Comment (dmohs) [reply src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs:94]: Is that necessary in this case? Locals are scoped to the component instance, after all.

File: src/cljs/main/broadfcui/common/components.cljs [hunk 4]
```diff
@@ -358,28 +358,25 @@
...
+  {:render
    (fn [{:keys [props]}]
-     (let [sep (icons/icon {:style {:color (:text-lightest style/colors) :padding "0 0.5rem"
-                                    :fontSize (:scale-factor props)}} :angle-right)
+     (let [sep (icons/icon {:style {:color (:text-lightest style/colors) :padding "0 0.5rem"}} :angle-right)
            crumbs (filter some? (:crumbs props))]
        (case (count crumbs)
          0 [:div {}]
-         1 [:div {} (:text (first crumbs))]
+         1 [:div {:style {:fontWeight "500" :fontSize "1.2em"}} (:text (first crumbs))]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common/components.cljs]: Nit: No quotes around font-weight.

File: src/cljs/main/broadfcui/common/icons.cljs [hunk 5]
```diff
@@ -32,6 +32,7 @@
    :close "fa-close"
    :reset "fa-undo"
    :new-window "fa-external-link"
+   :help "fa-question-circle"
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common/icons.cljs]: Nit: Maybe alphabetize since you're touching this.

File: src/cljs/main/broadfcui/main.cljs [hunk 6]
```diff
@@ -228,6 +229,13 @@
         [:div {:style {:width "100%" :borderBottom (str "1px solid " (:line-default style/colors))}}
          [:div {:style {:float "right" :fontSize "70%" :margin "0 0 0.5em 0"}}
           [AccountDropdown {:auth2 (:auth2 props)}]
+          [:a {:href (config/user-guide-url)
+               :target "_blank"
+               :style {:display "block" :float "right"
+                       :color (:text-light style/colors)
+                       :fontSize "1.2rem" :lineHeight "0.6rem"
+                       :padding "0.4rem" :marginRight "0.5rem"}}
+           (icons/icon {} :help)]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/main.cljs]: I feel like this needs some sort of hover text. It's not clear where it goes.

File: src/cljs/main/broadfcui/page/workspace/data/import_data.cljs [hunk 7]
```diff
@@ -32,7 +32,8 @@
                                (.readAsText reader (.slice file 0 preview-limit)))))}]
       [:div {:style {:marginBottom "0.8em"}} "For more information, see our "
        [:a {:href (config/user-guide-url) :target "_blank"} "user guide"] "."]
-      [comps/Button {:text "Choose file..." :onClick #(-> (@refs "entities") .click)}]
+      [comps/Button {:text (if (:upload-result @state) "Choose another file..." "Choose file...")
+                     :onClick #(-> (@refs "entities") .click)}]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/page/workspace/data/import_data.cljs:36]: After uploading a file that gave me an error, choosing another one didn't work. Probably not your fault but could you file the bug please?

File: src/cljs/main/broadfcui/page/workspace/monitor/tab.cljs [hunk 8]
```diff
@@ -10,6 +10,7 @@
     [broadfcui.nav :as nav]
     [broadfcui.page.workspace.monitor.common :as moncommon]
     [broadfcui.page.workspace.monitor.submission-details :as submission-details]
+    [broadfcui.persistence :as persistence]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/page/workspace/monitor/tab.cljs]: Looks like you didn't need this?

File: src/cljs/main/broadfcui/page/workspaces_list.cljs [hunk 9]
```diff
@@ -294,12 +294,13 @@
...
-       [:div {:style {:marginTop "2em"}}
+       [:div {:style {:marginTop "1.5rem"}}
         (if selected-ws-id
           [:div {}
-           [:div {:style {:margin "1em"}}
-            [:span {:style {:fontSize "180%"}}
-             [comps/Breadcrumbs {:scale-factor "60%" :crumbs (create-breadcrumbs-from-hash (:hash nav-context))}]]]
+           [:div {:style {:marginBottom "1.5rem" :paddingLeft "1.5rem" :fontSize "125%"}}
+            "Workspace: "
+            [:span {:style {:fontWeight "500"}}
```
Comment (dmohs) [inline src/cljs/main/broadfcui/page/workspaces_list.cljs]: I'm also okay with strings for these numbers, just be consistent.

**Review by dmohs — APPROVED** [review]
Very minor comments. Looks good.

File: src/cljs/main/broadfcui/page/workspace/data/import_data.cljs (hunk 7 above)
Comment (zarsky-broad) [reply src/cljs/main/broadfcui/page/workspace/data/import_data.cljs:36]: There's already a ticket for that, it's one of the children of the Kris doc

File: src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs (hunk 3 above)
Comment (zarsky-broad) [reply src/cljs/main/broadfcui/page/method_repo/method_config_importer.cljs:94]: Yeah, for that reason I think it doesn't need to change.

**Pull request merged**